computedfields/handlers.py

```diff
@@ -24,10 +25,51 @@
 STORAGE.DELETES = {}
 STORAGE.M2M_REMOVE = {}
 STORAGE.M2M_CLEAR = {}
+STORAGE.UPDATE_DIRTY = {}
 
 DELETES = STORAGE.DELETES
 M2M_REMOVE = STORAGE.M2M_REMOVE
 M2M_CLEAR = STORAGE.M2M_CLEAR
+UPDATE_DIRTY = STORAGE.UPDATE_DIRTY
+
+
+def update_dirty_handler(sender, instance, **kwargs):
+    """
+    ``update_dirty_handler`` handler.
+
+    ``pre_save`` signal handler to spot incoming fk relation changes.
+    This is needed to correctly update old relations after fk changes,
+    that contain dirty computed field values after a save.
+    The actual dirty updates are done during ``post_save``.
+    """
+    # do not handle fixtures
+    if kwargs.get('raw'):
+        return
+    modeldata = CFMT._map.get(sender)
+    # exit early if model is not part of any computed field depends chain
+    if not modeldata:
+        return
+    # exit early if save is a create
+    if instance.pk is None:
```

Comment:
Better to use `instance._state.adding` instead of `instance.pk is None`, as the latter isn't reliable (e.g. UUID primary keys have a pk even before saving).

Reply:
Ah nice, wasn't aware of that flag. :+1: